test/integration-tests/Uniswap.test.ts

```diff
@@ -1196,6 +1202,167 @@ describe('Uniswap V2 and V3 Tests:', () => {
+
+      planner.addCommand(CommandType.ERC721_PERMIT, [router.address, tokenId, MAX_UINT, v, r, s])
+
+      // bob permits the router to authorize token
+      await executeRouter(planner)
+
+      expect((await v3NFTPositionManager.positions(tokenId)).operator).to.eq(router.address)
+    })
+    it('decrease liquidity', async () => {
+      // first we need to permit the router to spend the nft
+      const { v, r, s } = await getPermitNFTSignature(bob, v3NFTPositionManager, router.address, tokenId, MAX_UINT)
+      planner.addCommand(CommandType.ERC721_PERMIT, [router.address, tokenId, MAX_UINT, v, r, s])
+
+      let position = await v3NFTPositionManager.positions(tokenId)
+      let liquidity = position.liquidity
+
+      const params = { tokenId: tokenId, liquidity: liquidity, amount0Min: 0, amount1Min: 0, deadline: MAX_UINT }
+
+      planner.addCommand(CommandType.V3_DECREASE_LIQUIDITY, [params])
+
+      await executeRouter(planner)
+
+      position = await v3NFTPositionManager.positions(tokenId)
+      liquidity = position.liquidity
+
+      expect(liquidity).to.eq(0)
+    })
+
+    it('collect', async () => {
+      // first we need to permit the router to spend the nft
+      let { v, r, s } = await getPermitNFTSignature(bob, v3NFTPositionManager, router.address, tokenId, MAX_UINT)
+      planner.addCommand(CommandType.ERC721_PERMIT, [router.address, tokenId, MAX_UINT, v, r, s])
+
+      let position = await v3NFTPositionManager.positions(tokenId)
+      let liquidity = position.liquidity
+
+      const decreaseParams = {
+        tokenId: tokenId,
+        liquidity: liquidity,
+        amount0Min: 0,
+        amount1Min: 0,
+        deadline: MAX_UINT,
+      }
+
+      planner.addCommand(CommandType.V3_DECREASE_LIQUIDITY, [decreaseParams])
+
+      await executeRouter(planner)
+
+      position = await v3NFTPositionManager.positions(tokenId)
+
+      let owed0 = position.tokensOwed0
+      let owed1 = position.tokensOwed1
+
+      planner = new RoutePlanner()
+
+      let bobBalanceBeforeToken0 = await usdcContract.balanceOf(bob.address)
+      let bobBalanceBeforeToken1 = await wethContract.balanceOf(bob.address)
+
+      const collectParams = { tokenId: tokenId, recipient: bob.address, amount0Max: owed0, amount1Max: owed1 }
+
+      planner.addCommand(CommandType.V3_COLLECT, [collectParams])
+
+      await executeRouter(planner)
+
+      let bobBalanceAfterToken0 = await usdcContract.balanceOf(bob.address)
+      let bobBalanceAfterToken1 = await wethContract.balanceOf(bob.address)
+
+      expect(bobBalanceAfterToken0.sub(bobBalanceBeforeToken0)).to.eq(owed0)
+      expect(bobBalanceAfterToken1.sub(bobBalanceBeforeToken1)).to.eq(owed1)
+
+      position = await v3NFTPositionManager.positions(tokenId)
+      owed0 = position.tokensOwed0
+      owed1 = position.tokensOwed1
+
+      expect(owed0).to.eq(0)
+      expect(owed1).to.eq(0)
+    })
+
+    it('burn', async () => {
```

Comment:
would love to see more test cases for each action. 
- can you burn multiple positions in the same tx?
- if you decrease your liquidity but dont collect, can a different user maliciously collect?
- what happens if someone who doesnt hold a position tries to burn someone elses position? like if A submits B's permit signature and tries to burn their token it shouldnt

any other attacks you can think of!